gitstatus.elv

```diff
@@ -228,8 +228,14 @@ fn query [repository]{
 
     echo $us$repository$rs > $state[stdin]
 
-    # this depends on Elvish commit 770904b, which introduces read-upto
-    response = (read-upto $rs < $state[stdout])
+    # read-upto depends on Elvish commit 770904b, otherwise fall back to calling bash
+    response = $nil
+    use builtin
+    if (has-key $builtin: read-upto~) {
```

Comment:
Out of curiosity, what does the tilde do here?

Reply:
This is how Elvish names the variables in which the lambda of a function is stored. The `read-upto` command lambda is stored in the `$read-upto~` variable:
```
> put $read-upto~
▶ <builtin read-upto>
```
This works for all functions. Since namespaces are accessible as hashes containing all the symbols in them, the `read-upto` function is stored in `$builtin:[read-upto~]`.